dist/sizzle.js

```diff
@@ -465,26 +465,51 @@ function createButtonPseudo( type ) {
  * @param {Boolean} disabled true for :disabled; false for :enabled
  */
 function createDisabledPseudo( disabled ) {
-	// Known :disabled false positives:
-	// IE: *[disabled]:not(button, input, select, textarea, optgroup, option, menuitem, fieldset)
-	// not IE: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
+
+	// Known :disabled false positives: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
 	return function( elem ) {
 
-		// Check form elements and option elements for explicit disabling
-		return "label" in elem && elem.disabled === disabled ||
-			"form" in elem && elem.disabled === disabled ||
+		// Only certain elements can match :enabled or :disabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-enabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-disabled
+		if ( "form" in elem ) {
+
+			// Check for inherited disabledness on relevant non-disabled elements:
+			// * listed form-associated elements in a disabled fieldset
+			//   https://html.spec.whatwg.org/multipage/forms.html#category-listed
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-fe-disabled
+			// * option elements in a disabled optgroup
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-option-disabled
+			// All such elements have a "form" property.
+			if ( elem.parentNode && elem.disabled === false ) {
+				return "label" in elem ?
```

Comment:
5dab06f307f716afbba720fedd097fdab5f7c36f already converted a ternary expression to logical statements; let me see what the hit is from introducing one or two more levels of nested `if`s. But yes, code like this is an example of why I don't like `no-nested-ternary`—every fork of this complex decision tree¹ has the single purpose of returning a boolean value indicating either enabledness or disabledness depending on outer function argument, and IMO that structure isn't rendered clearer by replacing expressions with statements. Since it's always going to be hairy, I'd rather optimize for size and handle comprehensibility with blank lines, indentation, and comments.

¹ in statement-based pseudocode:
```
if likely a listed form element:
  if disabledness might be inherited:
    if an option:
      if parent is an optgroup:
        parent.disabled represents disabledness
      else:
        elem.disabled represents disabledness
    else if elem.isDisabled is boolean:
      elem.isDisabled represents disabledness
    else:
      the existence of a disabled fieldset ancestor represents disabledness
  else:
    elem.disabled represents disabledness
else if likely an optgroup or menuitem:
  elem.disabled represents disabledness
else:
  neither enabled nor disabled
```